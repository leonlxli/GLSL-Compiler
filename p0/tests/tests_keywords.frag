void        
int         
float       
bool        
NULL        
[]          

<=          
>=          
==          
!=          

&&          
||          
while       
for         
if          
else        
return      

break       
layout      

++          
--          

vec2        
vec3        
vec4        
mat2        
mat3        
mat4        
ivec2       
ivec3       
ivec4       
bvec2       
bvec3       
bvec4       
in          
out         
inout       
layout      
uniform     
const       
continue    
do          
switch      
case        
default     
uint        
uvec2       
uvec3       
uvec4       
struct      

+ 
- 
* 
/ 
% 
< 
> 
= 
! 
; 
. 
, 
( 
) 
{ 
} 
[ 
] 