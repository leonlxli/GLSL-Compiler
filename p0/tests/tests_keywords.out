void         line 1 cols 1-4 is T_Void 
int          line 2 cols 1-3 is T_Int 
float        line 3 cols 1-5 is T_Float 
bool         line 4 cols 1-4 is T_Bool 
NULL         line 5 cols 1-4 is T_Null 
[]           line 6 cols 1-2 is T_Dims 
<=           line 8 cols 1-2 is T_LessEqual 
>=           line 9 cols 1-2 is T_GreaterEqual 
==           line 10 cols 1-2 is T_Equal 
!=           line 11 cols 1-2 is T_NotEqual 
&&           line 13 cols 1-2 is T_And 
||           line 14 cols 1-2 is T_Or 
while        line 15 cols 1-5 is T_While 
for          line 16 cols 1-3 is T_For 
if           line 17 cols 1-2 is T_If 
else         line 18 cols 1-4 is T_Else 
return       line 19 cols 1-6 is T_Return 
break        line 21 cols 1-5 is T_Break 
layout       line 22 cols 1-6 is T_Layout 
++           line 24 cols 1-2 is T_Inc 
--           line 25 cols 1-2 is T_Dec 
vec2         line 27 cols 1-4 is T_Vec2 
vec3         line 28 cols 1-4 is T_Vec3 
vec4         line 29 cols 1-4 is T_Vec4 
mat2         line 30 cols 1-4 is T_Mat2 
mat3         line 31 cols 1-4 is T_Mat3 
mat4         line 32 cols 1-4 is T_Mat4 
ivec2        line 33 cols 1-5 is T_Ivec2 
ivec3        line 34 cols 1-5 is T_Ivec3 
ivec4        line 35 cols 1-5 is T_Ivec4 
bvec2        line 36 cols 1-5 is T_Bvec2 
bvec3        line 37 cols 1-5 is T_Bvec3 
bvec4        line 38 cols 1-5 is T_Bvec4 
in           line 39 cols 1-2 is T_In 
out          line 40 cols 1-3 is T_Out 
inout        line 41 cols 1-5 is T_Inout 
layout       line 42 cols 1-6 is T_Layout 
uniform      line 43 cols 1-7 is T_Uniform 
const        line 44 cols 1-5 is T_Const 
continue     line 45 cols 1-8 is T_Continue 
do           line 46 cols 1-2 is T_Do 
switch       line 47 cols 1-6 is T_Switch 
case         line 48 cols 1-4 is T_Case 
default      line 49 cols 1-7 is T_Default 
uint         line 50 cols 1-4 is T_Uint 
uvec2        line 51 cols 1-5 is T_Uvec2 
uvec3        line 52 cols 1-5 is T_Uvec3 
uvec4        line 53 cols 1-5 is T_Uvec4 
struct       line 54 cols 1-6 is T_Struct 
+            line 56 cols 1-1 is '+' 
-            line 57 cols 1-1 is '-' 
*            line 58 cols 1-1 is '*' 
/            line 59 cols 1-1 is '/' 
%            line 60 cols 1-1 is '%' 
<            line 61 cols 1-1 is '<' 
>            line 62 cols 1-1 is '>' 
=            line 63 cols 1-1 is '=' 
!            line 64 cols 1-1 is '!' 
;            line 65 cols 1-1 is ';' 
.            line 66 cols 1-1 is '.' 
,            line 67 cols 1-1 is ',' 
(            line 68 cols 1-1 is '(' 
)            line 69 cols 1-1 is ')' 
{            line 70 cols 1-1 is '{' 
}            line 71 cols 1-1 is '}' 
[            line 72 cols 1-1 is '[' 
]            line 73 cols 1-1 is ']' 
